src/main/java/com/dao/DishDAO.java

```diff
@@ -0,0 +1,128 @@
+package com.dao;
+
+import com.model.Dish;
+
+import javax.sql.DataSource;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class DishDAO {
+
+    private static final String UPDATE_QUERY = "UPDATE Dishes SET dish=?, dish_type_id=?, price=? where id=?";
+    private static final String INSERT_QUERY = "INSERT INTO Dishes(id, dish, dish_type_id, price)  values (?,?,?,?)";
+    private static final String SELECT_QUERY = "SELECT * FROM Dishes WHERE id = ?";
+    private static final String DELETE_QUERY = "DELETE FROM Dishes WHERE id=?";
+    private static final String SELECT_ALL_QUERY = "SELECT * FROM Dishes";
+
+    private DataSource dataSource;
+    private static DishDAO instance;
+
+    private DishDAO() {
+    }
+
+    public static DishDAO getInstance() {
+        if (instance == null) {
+            instance = new DishDAO();
+        }
+        return instance;
+    }
+
+    public void setDataSource(DataSource dataSource) {
+        this.dataSource = dataSource;
+    }
+
+    public boolean create(Dish dish) {
+        boolean isCreated = false;
```

Comment:
Actually it's unnecessary to store values in this value. You can return true from try{} block and false from the and of the method

Reply:
made